packages/cli/src/commands/doctor/doctor.js

```diff
@@ -0,0 +1,131 @@
+import chalk from 'chalk';
+import envinfo from 'envinfo';
+import {logger} from '@react-native-community/cli-tools';
+import ISSUES from './issues';
+import {getLoader} from '../../tools/loader';
+import printFixOptions, {KEYS} from './printFixOptions';
+import runAutomaticFix, {AUTOMATIC_FIX_LEVELS} from './runAutomaticFix';
+
+const printCategory = ({label, key}) => {
+  if (key > 0) {
+    logger.log();
+  }
+
+  logger.log(chalk.dim(label));
+};
+
+const printIssue = ({label, needsToBeFixed, isRequired}) => {
+  const symbol = needsToBeFixed
+    ? isRequired
+      ? chalk.red('✖')
+      : chalk.yellow('●')
+    : chalk.green('✓');
+
+  logger.log(` ${symbol} ${label}`);
+};
+
+export default (async function runDoctor() {
+  const Loader = getLoader();
+  const loader = new Loader();
+
+  loader.start('Running diagnostics...');
+
+  const environmentInfo = JSON.parse(
+    await envinfo.run(
+      {
+        Binaries: ['Node', 'Yarn', 'npm', 'Watchman'],
+        IDEs: ['Xcode', 'Android Studio'],
+        SDKs: ['iOS SDK', 'Android SDK'],
+        npmPackages: ['react', 'react-native', '@react-native-community/cli'],
+        npmGlobalPackages: ['*react-native*'],
+      },
+      {json: true, showNotFound: true},
+    ),
+  );
+
+  const iterateOverIssues = async ({label, issues}) => ({
+    label,
+    issues: (await Promise.all(
```

Comment:
Nice use of Promise.all to run all the checks at the same time, this is heavily I/O bound so this should be helpful